ably/ably-chat-js
# react-hooks: Add useTyping hook.

### Context

* [CHA-386]
* As part of the react hooks development, a new hook to interact with the `Typing` instance of a room was required.

### Description

- Added the `useTyping` hook to provide access to the `Typing` instance and associated functionality.
  -  Start and stop events can be triggered by calling the methods returned from the hook.
  - A listener can also be provided to receive any incoming events that the `Typing` instance of the room receives.
  - The hook returns a state, that it keeps up-to-date, containing the set of current typers.
 
- Updated the demo app to use the new hook, removed old functionality.
- Added a section on this hooks usage to the README.md.

### Checklist


### Testing Instructions (Optional)

1. Spin up a local instance of the demo app.
2. Open up two tabs and start typing on one.
3. Check a typing indicators appears and then disappears after you stop typing. (Should be a short delay) 

[CHA-386]: https://ably.atlassian.net/browse/CHA-386?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
